app/main.py

```diff
@@ -16,4 +16,17 @@
 }
 collection_of_coins = {1, 2, 25}
 
-# write your code here
+
+def sort_variables_by_mutability(*args) -> dict:
+    sorted_variables = {"mutable": [], "immutable": []}
+
+    for arg in args:
+        if isinstance(arg, (list, set, dict)):
+            sorted_variables["mutable"].append(arg)
+        else:
+            sorted_variables["immutable"].append(arg)
```

Comment:
The way you're checking mutability works for the types you've considered (list, set, dict), but there are other mutable types in Python, like user-defined classes, or even some built-in types like bytearray. A more reliable way to check for mutability could be to try to change the value and see if it raises an exception. However, be aware that this is not a perfect solution, as some objects may be mutable in some ways and immutable in others (for example, you can modify an individual byte of a bytearray, but you can't change its length).